numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -379,11 +379,16 @@ array_implement_c_array_function_creation(
         return Py_NotImplemented;
     }
 
-    PyObject *relevant_args = PyTuple_Pack(1,
-            PyDict_GetItem(kwargs, npy_ma_str_like));
-    if (relevant_args == NULL) {
+    PyObject *like_arg = PyDict_GetItem(kwargs, npy_ma_str_like);
+    if (like_arg == NULL) {
         return NULL;
     }
+    else if (!get_array_function(like_arg)) {
+        return PyErr_Format(
+            PyExc_TypeError,
```

Comment:
```suggestion
        return PyErr_Format(PyExc_TypeError,
```
We usually put the exception on the same line.  (I never realize that these function return NULL, but that looks nice).